In [1]:
import requests
from bs4 import BeautifulSoup

url = 'https://comic.naver.com/webtoon/weekdayList.nhn?week=mon'

resp = requests.get(url)

soup = BeautifulSoup(resp.content, 'lxml')
title_tag = soup.select('ul.img_list li dl dt a')
author_tag = soup.select('ul.img_list li dl dd a')
rating_tag = soup.select('ul.img_list li dl dd strong')

len(rating_tag)

51

In [2]:
title_tag

[<a href="/webtoon/list.nhn?titleId=733766&amp;weekday=mon" title="인생존망">인생존망</a>,
 <a href="/webtoon/list.nhn?titleId=183559&amp;weekday=mon" title="신의 탑">신의 탑</a>,
 <a href="/webtoon/list.nhn?titleId=648419&amp;weekday=mon" title="뷰티풀 군바리">뷰티풀 군바리</a>,
 <a href="/webtoon/list.nhn?titleId=602910&amp;weekday=mon" title="윈드브레이커">윈드브레이커</a>,
 <a href="/webtoon/list.nhn?titleId=654774&amp;weekday=mon" title="소녀의 세계">소녀의 세계</a>,
 <a href="/webtoon/list.nhn?titleId=733074&amp;weekday=mon" title="백수세끼">백수세끼</a>,
 <a href="/webtoon/list.nhn?titleId=728750&amp;weekday=mon" title="장씨세가 호위무사">장씨세가 호위...</a>,
 <a href="/webtoon/list.nhn?titleId=730174&amp;weekday=mon" title="칼가는 소녀">칼가는 소녀</a>,
 <a href="/webtoon/list.nhn?titleId=739115&amp;weekday=mon" title="앵무살수">앵무살수</a>,
 <a href="/webtoon/list.nhn?titleId=709731&amp;weekday=mon" title="유일무이 로맨스">유일무이 로맨스</a>,
 <a href="/webtoon/list.nhn?titleId=597478&amp;weekday=mon" title="평범한 8반">평범한 8반</a>,
 <a href="/webtoon/list.nhn?titleId=716857&amp

In [0]:
data_list = []
for title, author, rating in zip(title_tag, author_tag, rating_tag):
    data = {
        'title' : title.text,
        'author' : author.text,
        'rating' : rating.text,
    }
    data_list.append(data)

In [5]:
import copy 

data_list

webtoon_list = copy.deepcopy(data_list)
webtoon_list

[{'author': '박태준 / 전선욱', 'rating': '9.81', 'title': '인생존망'},
 {'author': '전체보기', 'rating': '9.94', 'title': '신의 탑'},
 {'author': 'SIU', 'rating': '9.82', 'title': '뷰티풀 군바리'},
 {'author': '전체보기', 'rating': '9.89', 'title': '윈드브레이커'},
 {'author': '설이 / 윤성원', 'rating': '9.97', 'title': '소녀의 세계'},
 {'author': '전체보기', 'rating': '9.87', 'title': '백수세끼'},
 {'author': '조용석', 'rating': '9.97', 'title': '장씨세가 호위...'},
 {'author': '전체보기', 'rating': '9.97', 'title': '칼가는 소녀'},
 {'author': '모랑지', 'rating': '9.97', 'title': '앵무살수'},
 {'author': '전체보기', 'rating': '9.99', 'title': '유일무이 로맨스'},
 {'author': '치즈', 'rating': '9.85', 'title': '평범한 8반'},
 {'author': '전체보기', 'rating': '9.99', 'title': '오늘의 순정망화'},
 {'author': '조형근 / 김인호', 'rating': '9.92', 'title': '서브 콤플렉스'},
 {'author': '전체보기', 'rating': '9.82', 'title': '데드라이프'},
 {'author': '오리', 'rating': '9.95', 'title': '야자괴담'},
 {'author': '전체보기', 'rating': '9.91', 'title': '니편내편'},
 {'author': '김성진', 'rating': '9.98', 'title': '귀전구담'},
 {'author': '

과제1. author가 전체보기로 되어있는 데이터는 webtoon_list에서 모두 삭제해주세요.

과제2. rating이 9.9 이상인 웹툰의 제목 따로 recommended 변수에 담아주세요.

과제3(추가과제). rating이 제일 높은 데이터 3개만 top_recommended에 담아주세요.